docs/protocol_upgrade.md

```diff
@@ -69,6 +69,12 @@ In practice, we have all nightly protocol features enabled for Nayduck tests and
 
 ### Feature Stabilization
 
-When we are ready to make a new release, we examine which features should be stabilized and for those that should be, we
-remove the rust feature gating everywhere.
-In addition, we increment the `PROTOCOL_VERSION` constant and change the `protocol_version` implementation to map the stabilized features to the new protocol version.
+New protocol features are introduced first as nightly features and when the author of the feature thinks that
+the feature is ready to be stabilized, they should submit a pull request to stabilize the feature using
+[this template](../.github/PULL_REQUEST_TEMPLATE/feature_stabilization.md).
+In this pull request, they should the feature gating, increase the `PROTOCOL_VERSION` constant,
+and change the `protocol_version` implementation to map the stabilized features to the new protocol version.
+
+A feature stabilization request must be approved by at least **two** [nearcore code owners](https://github.com/orgs/near/teams/nearcore-codeowners)
+Unless it is a security related fix, a protocol feature cannot be released until at least two weeks after its stabilization.
```

Comment:
I wonder how we could 'test' it during those 2 weeks? should we make sure that someone explicitly tests something on betanet?

Reply:
I think this ambiguous. Does this say "don't send stabilization PR against release candidates from testnet" or does this say "don't send stabilization PR in the two weeks before the testnet release is cut"?

As in, do four weeks on a testnet count as "two weeks before release"? Is release a "testnet release" or "mainet release"?

I am also unsure what rule we *should* have here. I *think* what we want to prevent is people rushing stabilizations just before release is cut, as is a common practice today. In this sense, we want the second formulation from above. However, there's an angle that stabilization should be "completely" orthogonal from the release schedule. Ideally, when you submit a stabilization PR, you shouldn't be thinking about releases at all...

Reply:
> Ideally, when you submit a stabilization PR, you shouldn't be thinking about releases at all...

Yes, this is really the issue. And it has nothing to do with protocol features either.  For example, I think some (maybe all) of the changes related to num_epoch_to_store and gc_fork_clean_step shouldn’t have made the cut.  If your feature did not make it into master the day of the release, it doesn’t get to be in the release.  And if you send a PR five weeks before the next release but it took 36 days to review, you’re out of luck.